manifests/init.pp

```diff
@@ -80,6 +83,7 @@
   String $package_source                          = $nsclient::params::package_source,
   String $package_name                            = $nsclient::params::package_name,
   $download_destination                           = $nsclient::params::download_destination,
+  String $config_content                          = '',
```

Comment:
instead of empty string, could you default to undef?
```suggestion
  Optional[String[1]] $config_content             = undef,
```

That is the current best practice. For more information, you can checkout our guidelines: https://voxpupuli.org/docs/#reviewing-a-module-pr